code/modules/integrated_electronics/core/tools.dm

```diff
@@ -240,6 +240,99 @@
 
 
 
+/obj/item/device/integrated_electronics/analyzer
+	name = "circuit analyzer"
+	desc = "This tool can scan assembly and give code ,that is neeed to recreate this assembly in circuit printer."
```

Comment:
`This tool can scan an assembly in generate code necessary to recreate it in a circuit printer.`